### PR TITLE
added state check for state/:id endpoint

### LINKED ISSUE #3132: [Dev]State endpoint does not check user permissions
Feature -  
The endpoint of '/states/{id}' does not seem to be checking the user's permissions to see if they are authorized for that state.  As a result if I am the state administrator for Alaska and I request /states/az I get the results for Arizona.  I believe the response should be 403 unauthorized in this instance.  I am putting this in as a feature and not a bug because there is no indication in the code that this is the intended behavior.  The endpoint tests don't show a test for this and the code does not appear to be checking.

### This task is done when…
 - [ ] When a request is made for a user that does not have access to a state they receive a 403 unauthorized response.

### PR DESCRIPTION
**Pull request by knollfear**
resolves #3132

**Description-**
Add check on state/id endpoint to ensure that the user has access to this information
**This pull request changes...**

- adds middleware to the state/:id endpoint

**This pull request also touches…**

- possible side effects of this change
unlikely.  This simply introduces an additional check to the existing state/:id endpoint
**This pull request was tested in the follow ways…**

- list of frontend cases
- list of backend cases

**Steps to manually verify this change...**

1. This can currently only be manually verified via a postman test
2. log in to the website and capture your token (it is in the cookies)
3. open postman and make a get request to <URL>/states/ak (or any state that makes sense given your token).  You must use your jwt in the authorization header which will say "Bearer <YOUR_JWT>
4. confirm that you get the right answer
5. run the test again with a different state and get a validation error
6. test again with fed-admin user that has access to all states

### This pull request is ready to review when...

- [ ] Automated tests are updated (and all tests are passing)
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: api/routes/states/get.endpoint.js [hunk 1]
```diff
@@ -23,18 +23,31 @@ describe('US States endpoint', () => {
...
-    it('returns 404', async () => {
+    it('returns 403', async () => {
       const authedClient = login();
+      // This state does not exist, therefore it can't be available to this user.
+      const response = await authedClient.get('/states/zz');
+      expect(response.status).toEqual(403);
+    });
+    it('works for a Federal Admin', async () => {
+      const authedClient = login('fed-admin');
+      // This state does not exist, therefore it can't be available to this user.
```
Comment (tbolt) [inline api/routes/states/get.endpoint.js:44]: I think this is an extra comment that snuck in here.

**Review by tbolt — APPROVED** [review]
Looks good and all tests pass.

Tested with an http client:
1. Logged in with norole user and manually added AL state contractor permissions
2. hit /states/al and it returned 200 with expected contents
3. hit /states/ak and it returned 403
4. Logged in with fedadmin
5. hit /states/al and it returned 200 with expected contents
6. hit /states/zz an it returned 404 not found

File: api/routes/states/get.endpoint.js (hunk 1 above)
Comment (knollfear) [reply api/routes/states/get.endpoint.js]: Removed it.

**Comment by thetif** [comment]
looks good to me too. Tested with suggestions above and using norole and requesting access to CO and and trying it with only requested access and it returned 200 and expected contents.

**Comment by thetif** [comment]
@knollfear can you add a line to the changelog? you can put it in backend changes

**Review by jeromeleecms — APPROVED** [review]
Tested as directed in PR description. Fed admin yielded status 200 for all states. Generic login presented with 403 error when accessing non authorized states.

Looks good -- @thetif seems like Mike put in the changes requested. Anything else on this?

**Pull request merged**